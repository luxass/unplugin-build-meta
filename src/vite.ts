import { createVitePlugin } from "unplugin";
import { unpluginFactory } from "./";

/**
 * Vite plugin
 *
 * @example
 * ```ts
 * // vite.config.ts
 * import buildMeta from "unplugin-build-meta/vite"
 *
 * export default defineConfig({
 *   plugins: [buildMeta()],
 * })
 * ```
 */
export default createVitePlugin(unpluginFactory);
