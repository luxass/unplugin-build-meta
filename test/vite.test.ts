import { join } from "node:path";
import { build } from "vite";
import { describe, expect, it } from "vitest";
import { testdir } from "vitest-testdirs";
import buildMeta from "../src/vite";

describe("handles git metadata", () => {
  it("expect git metadata to be available", async () => {
    const testdirPath = await testdir.from(join(import.meta.dirname, "fixtures/git"));

    expect(testdirPath).toBeDefined();

    const result = await build({
      build: {
        lib: {
          entry: join(testdirPath, "git.js"),
          formats: ["es"],
          fileName: "bundle",
          name: "bundle",
        },
        outDir: join(testdirPath, "dist"),
        minify: false,
      },
      plugins: [buildMeta({
        modules: ["git"],
      })],
    });

    if (!Array.isArray(result)) {
      expect.fail("result is not an array");
    }

    expect(result).toBeDefined();

    const firstResult = result[0];

    expect(firstResult).toBeDefined();
    expect(firstResult?.output).toBeDefined();
    expect(firstResult?.output[0]).toBeDefined();
    expect(firstResult?.output[0].code).toBeDefined();
    expect(firstResult?.output[0].code).toMatchSnapshot();
  });

  it("expect specific git properties to be importable", async () => {
    const testdirPath = await testdir.from(join(import.meta.dirname, "fixtures/git"));

    expect(testdirPath).toBeDefined();

    const result = await build({
      build: {
        lib: {
          entry: join(testdirPath, "git-specific.js"),
          formats: ["es"],
          fileName: "bundle",
          name: "bundle",
        },
        outDir: join(testdirPath, "dist"),
        minify: false,
      },
      plugins: [buildMeta({
        modules: ["git"],
      })],
    });

    if (!Array.isArray(result)) {
      expect.fail("result is not an array");
    }

    expect(result).toBeDefined();

    const firstResult = result[0];

    expect(firstResult).toBeDefined();
    expect(firstResult?.output).toBeDefined();
    expect(firstResult?.output[0]).toBeDefined();
    expect(firstResult?.output[0].code).toBeDefined();
    expect(firstResult?.output[0].code).toMatchSnapshot();
  });
});
