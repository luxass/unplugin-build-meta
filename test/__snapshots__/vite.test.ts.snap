// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`handles git metadata > expect git metadata to be available 1`] = `
"const repositoryUrl = "https://github.com/luxass/unplugin-build-meta";
const sha = "0d26380e72f1cc40efa0388abe3e2e998db9d772";
const shortSha = "0d26380e72";
const branch = "main";
const tag = null;
const tags = [];
const lastTag = null;
const commitAuthorName = "Lucas";
const commitAuthorEmail = "lucasnrgaard@gmail.com";
const commitAuthorDate = "2025-04-16T16:53:03+02:00";
const commitCommitterName = "Lucas";
const commitCommitterEmail = "lucasnrgaard@gmail.com";
const commitCommitterDate = "Wed Apr 16 16:53:03 2025 +0200";
const latestCommitMessage = "chore: fix eslint ignore";
const git = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  branch,
  commitAuthorDate,
  commitAuthorEmail,
  commitAuthorName,
  commitCommitterDate,
  commitCommitterEmail,
  commitCommitterName,
  lastTag,
  latestCommitMessage,
  repositoryUrl,
  sha,
  shortSha,
  tag,
  tags
}, Symbol.toStringTag, { value: "Module" }));
console.log(git);
"
`;

exports[`handles git metadata > expect specific git properties to be importable 1`] = `
"const sha = "0d26380e72f1cc40efa0388abe3e2e998db9d772";
const shortSha = "0d26380e72";
const branch = "main";
console.log({ branch, sha, shortSha });
"
`;
